opesci/staggeredgrid.py

```diff
@@ -915,12 +922,15 @@ def stress_loop(self):
                         + ccode(kernel.xreplace({self.t+1: self.time[1], self.t: self.time[0]})) + ';\n'
             dict1 = {'i': i, 'i0': i0, 'i1': i1, 'body': body}
             body = render(tmpl, dict1)
-            if self.ivdep and d == self.dimension-1:
+            if not self.pluto and self.ivdep and d == self.dimension-1:
                     body = '%s\n' % self.compiler._ivdep + body
-            if self.simd and d == self.dimension-1:
+            if not self.pluto and self.simd and d == self.dimension-1:
                     body = '#pragma simd\n' + body
 
-        if self.omp:
+        if self.pluto:
+            body = '#pragma scop\n'+body
+            body += '#pragma endscop'
+        elif self.omp:
```

Comment:
I did this because  we originally added "#pragma omp for " if condition "self.omp"is true, so I have to say if self.pluto , don't add  "#pragma omp for " because it will break pluto. So I think it will be easier to manage pluto's "scop" here as well.

Date: Fri, 11 Sep 2015 08:28:22 -0700
From: notifications@github.com
To: opesci-fd@noreply.github.com
CC: zlcghoul@hotmail.com
Subject: Re: [opesci-fd] Pluto (#45)

In opesci/staggeredgrid.py:

>                      body = '#pragma simd\n' + body
>  
> -        if self.omp:
> +        if self.pluto:
> +            body = '#pragma scop\n'+body
> +            body += '#pragma endscop'
> +        elif self.omp:

Ok, agreed let's start with exploring spatial tiling on a single loop. Still can you please move the scop pragmas to the template itself (guarded by an %if pluto==True). This way it is easier to identify the tiling regions for experimentation and move them if required.


―
Reply to this email directly or view it on GitHub.